tests/unit/customizations/s3/test_subcommands.py

```diff
@@ -132,6 +132,26 @@ def test_ls_command_with_no_args(self):
             's3', region_name=None, endpoint_url=None, verify=None,
             config=None))
 
+    def test_ls_command_with_non_existsent_url(self):
+        ls_command = ListCommand(self.session)
+        parsed_global = FakeArgs(region=None, endpoint_url=None,
+                                    verify_ssl=None)
+        parsed_args = FakeArgs(dir_op=False, paths='s3:///random_nonsense',
+                                human_readable=False, summarize=False,
+                                request_payer=None, page_size=None)
+        ls_command._run_main(parsed_args, parsed_global)
+        # We should make no operation calls.
+        call = self.session.create_client.return_value.list_buckets
+        self.assertEqual(call.call_count, 0)
+
+        # Verify get_client
+        get_client = self.session.create_client
+        args = get_client.call_args
+        
```

Comment:
There seems to be some extra whitespace appended on this line. Could we get this removed?